### PR TITLE
[Forge] Configure nightly run to run for 4 hours

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

Configure the nightly forge run to run for 4 hours to gather more reliability signals. 

### Test Plan

Forge run

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2451)

### CODE REVIEW

File: testsuite/forge-test-runner-template.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
     - -c
     - |
       ulimit -n 1048576
-      forge --suite {FORGE_TEST_SUITE} test k8s-swarm --image-tag {IMAGE_TAG} --namespace {FORGE_NAMESPACE} {REUSE_ARGS} {KEEP_ARGS} {ENABLE_HAPROXY_ARGS}
+      forge --suite {FORGE_TEST_SUITE} --duration {FORGE_RUNNER_DURATION_SECS} --avg-tps {FORGE_RUNNER_TPS_THRESHOLD} test k8s-swarm --image-tag {IMAGE_TAG} --namespace {FORGE_NAMESPACE} {REUSE_ARGS} {KEEP_ARGS} {ENABLE_HAPROXY_ARGS}
```
Comment (rustielin) [inline testsuite/forge-test-runner-template.yaml:21]: `--duration-secs` ?

File: .github/workflows/run-forge.yaml [hunk 2]
```diff
@@ -32,6 +32,8 @@ env:
   FORGE_COMMENT: forge_comment.txt
   FORGE_PRE_COMMENT: forge_pre_comment.txt
   FORGE_RUNNER_MODE: k8s
+  FORGE_RUNNER_DURATION_SECS: 300
```
Comment (rustielin) [inline .github/workflows/run-forge.yaml:35]: you will also need to declare these in the `inputs` section above. This allows other workflows to call this workflow and specify these variables

File: testsuite/forge-test-runner-template.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ spec:
     - -c
     - |
       ulimit -n 1048576
-      forge --suite {FORGE_TEST_SUITE} test k8s-swarm --image-tag {IMAGE_TAG} --namespace {FORGE_NAMESPACE} {REUSE_ARGS} {KEEP_ARGS} {ENABLE_HAPROXY_ARGS}
+      forge --suite {FORGE_TEST_SUITE} --duration-secs {FORGE_RUNNER_DURATION_SECS} --avg-tps {FORGE_RUNNER_TPS_THRESHOLD} test k8s-swarm --image-tag {IMAGE_TAG} --namespace {FORGE_NAMESPACE} {REUSE_ARGS} {KEEP_ARGS} {ENABLE_HAPROXY_ARGS}
```
Comment (rustielin) [inline testsuite/forge-test-runner-template.yaml:21]: You'll also need to template `FORGE_RUNNER_DURATION_SECS` and `FORGE_RUNNER_TPS_THRESHOLD` in via the `sed` command on line 214 of `run_forge.sh`

File: .github/workflows/run-forge.yaml [hunk 4]
```diff
@@ -32,6 +37,8 @@ env:
...
```
Comment (rustielin) [inline .github/workflows/run-forge.yaml:40]: you probably mean to do `${{ inputs. FORGE_RUNNER_DURATION_SECS }}`

File: testsuite/run_forge.sh [hunk 5]
```diff
@@ -184,8 +185,8 @@ if [ "$FORGE_RUNNER_MODE" = "local" ]; then
     # more file descriptors for heavy txn generation
     ulimit -n 1048576
 
-    cargo run -p forge-cli -- --suite $FORGE_TEST_SUITE --mempool-backlog 1500 --avg-tps $LOCAL_TPS_THRESHOLD \
-        --max-latency-ms $LOCAL_P99_LATENCY_MS_THRESHOLD \
+    cargo run -p forge-cli -- --suite $FORGE_TEST_SUITE --workers-per-ac 10 --avg-tps $FORGE_RUNNER_TPS_THRESHOLD \
+        --max-latency-ms $LOCAL_P99_LATENCY_MS_THRESHOLD --duration-secs $FORGE_RUNNER_DURATION_SECS  \
```
Comment (rustielin) [inline testsuite/run_forge.sh:189]: nit: should we also get rid of `LOCAL_P99_LATENCY_MS_THRESHOLD` in favor of the more generic e.g. `FORGE_RUNNER_P99_LATENCY_MS_THRESHOLD` instead?

**Review by rustielin — APPROVED** [review]
LGTM % the nit.

We'll still need to change the auto clean-up logic to prevent these long-running Forge jobs from getting killed tho

File: testsuite/run_forge.sh (hunk 5 above)
Comment (sitalkedia) [reply testsuite/run_forge.sh:189]: Yes, we can do that as a follow up - the latency metrics is something I will need to work on revamping. Currently we are measuring p99 e2e latency, which is noisy and also not sufficient.

**Pull request merged**